client/app/components/packages/pas-form/edit.hbs

```diff
@@ -19,8 +19,9 @@
         <strong>Project Name</strong>
         <Input
           @type="text"
-          @value={{@package.pasForm.dcpRevisedprojectname}}
-          data-test-dcprevisedprojectname
+          @value={{if @package.pasForm.dcpRevisedprojectname @package.pasForm.dcpRevisedprojectname @package.project.dcpProjectname}}
+          @key-down={{fn (mut @package.pasForm.dcpRevisedprojectname)}}
```

Comment:
I'm getting a little nervous about relying on these key-down events. How does this affect people who might right-click and paste data into the field? Wouldn't that not trigger a key-down event?

I think you should probably _bind_ the specific value here:

```
@value={{@package.pasForm.dcpRevisedprojectname}}
```

THEN in your pas-form model, add a [defaultValue](https://api.emberjs.com/ember-data/3.18/functions/@ember-data%2Fmodel/attr):

```
@attr({
  defaultValue(pasForm) {
    return pasForm.package.dcpProjectname;
  },
})
dcpRevisedprojectname;
```

See this [blog post](https://davidtang.io/2018/05/26/conditional-default-values-for-ember-data-model-attributes.html) for more examples.

It's unclear to me whether this defaultValue gets triggered when the server provides an empty string. Happy to jump on a call and work through that.

More broadly, though, my reasoning here is that this behavior is pretty fundamental to the business logic itself. Instead of keeping that critical behavior inside the if helper in a template, it's more apparent on the model level.

Reply:
that's a good point, will change